demo/cts_poc/notary-sign.sh

```diff
@@ -0,0 +1,113 @@
+#!/bin/bash
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+### Simple script to generate a container image signature to submit to a SCITT ledger using Notation and a self-signed certificate in Azure Key Vault.
+### The script assumes that the user is already logged in to Azure, has proper access and permissions to the given Key Vault and Contaner Registry.
+### It is also assumed that the referenced container image is already present in the given Container Registry and that the certificate is present in the given Key Vault.
+### Please make sure that the certificate complies with the Notary certificate requirements: https://github.com/notaryproject/specifications/blob/v1.0.0/specs/signature-specification.md#certificate-requirements
+
+set -e
+
+# Variables
+: "${AKV_NAME:?"variable not set. Please define the Azure Key Vault name where the self-signed certificate is located"}"
+: "${CERTIFICATE_NAME:?"variable not set. Please define the name of the self-signed certificate in Azure Key Vault"}"
+: "${CERTIFICATE_VERSION:?"variable not set. Please define the version of the self-signed certificate in Azure Key Vault"}"
+: "${ACR_NAME:?"variable not set. Please define the name of the Azure Container Registry where the container image to sign should be found"}"
+: "${IMAGE_REPOSITORY:?"variable not set. Please define the image repository in the Azure Container Registry where the container image to sign should be found"}"
+: "${SIGNATURE_OUTPUT_PATH:?"variable not set. Please define the path where the COSE signature will be saved to"}"
+
+# Optional variables
+IMAGE_TAG=${IMAGE_TAG:-""}
+IMAGE_DIGEST=${IMAGE_DIGEST:-""}
+
+# Optional variables for Notation signing
+# Specify true or false whether the certificate in AKV is a self-signed one or not
+IS_SELF_SIGNED_CERT=${IS_SELF_SIGNED_CERT:-""}
+# Specify the path to the CA certificates PEM file if the certificate in AKV does not contain the full certificate chain
+CA_CERTS_PEM_FILE_PATH=${CA_CERTS_PEM_FILE_PATH:-""}
+
+# Check if the image tag or digest is provided
+if [ -z "$IMAGE_TAG" ] && [ -z "$IMAGE_DIGEST" ]; then
+    echo "Either IMAGE_TAG or IMAGE_DIGEST must be provided"
+    exit 1
+fi
+
+echo -e "\nInstall ORAS CLI"
+
+# Install ORAS CLI
+ORAS_VERSION="1.1.0"
+curl -LO "https://github.com/oras-project/oras/releases/download/v${ORAS_VERSION}/oras_${ORAS_VERSION}_linux_amd64.tar.gz"
```

Comment:
It is possible to simplify every untar operation here, e.g.
```
curl -sL "https://github.com/oras-project/oras/releases/download/v1.1.0/oras_1.1.0_linux_amd64.tar.gz" | tar xz
mv oras /usr/local/bin/
```

Reply:
Addressed in https://github.com/microsoft/scitt-ccf-ledger/pull/172/commits/d5e04ded76fb314f21e9e20ace2e7bf10d0f7570